test/Core.Test/Services/OrganizationServiceTests.cs

```diff
@@ -649,6 +648,78 @@ await sutProvider.GetDependency<IMailService>().Received(1)
         await sutProvider.Sut.SaveUserAsync(newUserData, savingUser.UserId, collections, groups);
     }
 
+    [Theory, BitAutoData]
+    public async Task SaveUser_WithCustomPermission_WhenCustomSavingUserHasPermission_Passes(
+        Organization organization,
+        [OrganizationUser(type: OrganizationUserType.User)] OrganizationUser oldUserData,
+        [OrganizationUser(type: OrganizationUserType.Custom)] OrganizationUser newUserData,
+        IEnumerable<CollectionAccessSelection> collections,
+        IEnumerable<Guid> groups,
+        [OrganizationUser(type: OrganizationUserType.Custom)] OrganizationUser savingUser,
```

Comment:
You also need to make the `savingUser` a custom user type, otherwise you might hit the early returns (if admin or owner) and get a false positive. (Assuming that the user role is random, maybe it defaults to something sensible)

Reply:
oh I missed that, it's even in the part that I commented on 😅